components/progress-bar/src/ProgressBar.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { CSSProperties, useEffect, useState } from "react";
+import { useProgress } from "@locoworks/reusejs-toolkit-react-hooks";
+import { twMerge } from "tailwind-merge";
+import { type } from "os";
+
+interface ProgressBarInterface {
+  progressInterval: number;
+  defaultProgress: number;
+  running: boolean;
+  totalFileSize?: number;
+  processedFileSize?: number;
+  progressClasses?: string | CSSProperties;
+  progressContainerClasses?: string | CSSProperties;
+  progressText?: string;
+  progressTextClasses?: string | CSSProperties;
+  radius?: number;
+  circleRadiousInPercentage?: string;
+  enableCircularProgress?: boolean;
+  StrokeColour?: string;
+  progressColour?: string;
+}
+
+const ProgressBar = ({
+  progressClasses,
+  progressContainerClasses,
+  progressInterval,
+  running,
+  defaultProgress,
+  totalFileSize = 0,
+  processedFileSize = 0,
+  progressText,
+  progressTextClasses,
+  radius = 95,
+  circleRadiousInPercentage,
+  enableCircularProgress,
+  StrokeColour,
+  progressColour,
+}: ProgressBarInterface) => {
+  const defaultProgressClasses = "flex h-full rounded-full bg-green-600";
+  const defaultProgressContainerClasses =
+    "flex h-2 rounded-md mx-1 w-full rounded-full overflow-hidden";
+
+  const [progressStyle, setProgressStyle] = useState({});
+  const [statusStyle, setStatusStyle] = useState({});
+
+  const { progress } = useProgress({
+    progressInterval,
+    running,
+    defaultProgress,
+    totalFileSize,
+    processedFileSize,
+  });
+
+  useEffect(() => {
+    const circumference = 2 * Math.PI * radius;
+    const offset = circumference - (progress / 100) * circumference;
+
+    setProgressStyle({
+      strokeDasharray: `${circumference} ${circumference}`,
+      strokeDashoffset: offset,
+    });
+
+    setStatusStyle({
+      strokeDasharray: `${circumference} ${circumference}`,
+      strokeDashoffset: 0,
+    });
+  }, [progress]);
+
+  return (
+    <>
+      {enableCircularProgress ? (
+        <div
+          className={`h-full w-full flex justify-center items-center ${
+            typeof progressContainerClasses === "string"
+              ? twMerge(
+                  defaultProgressContainerClasses,
+                  progressContainerClasses
+                )
+              : defaultProgressContainerClasses
+          }`}
+          style={
+            typeof progressContainerClasses === "object"
+              ? progressContainerClasses
+              : {}
+          }
+        >
+          <svg
+            className={`w-52 h-52 rounded-full ${
+              typeof progressClasses === "string" ? progressClasses : ""
```

Comment:
CSSProperties type of input are not handles here.

Reply:
it is handled